packages/cascara/src/system-components/Boundaries/fallbacks/ErrorFallback.js

```diff
@@ -8,18 +9,44 @@ const propTypes = {
 };
 
 const ErrorFallback = ({ error, resetErrorBoundary }) => {
+  const checkedFromLS =
+    window.localStorage.getItem('ESPRESSIVE_SHOW_ERRORS') === 'true';
+  const [checked, setChecked] = useState(checkedFromLS);
+  const showError = process.env.NODE_ENV === 'development' || checked;
+
+  const hangleCheckboxClick = useCallback(() => {
+    setChecked(!checked);
+  }, [checked]);
+
+  const hangleTryAgainClick = useCallback(() => {
+    window.localStorage.setItem('ESPRESSIVE_SHOW_ERRORS', checked);
+
+    if (resetErrorBoundary) {
+      resetErrorBoundary();
+    }
+  }, [checked, resetErrorBoundary]);
+
   return (
-    <div className='ui tiny error message' role='alert'>
+    <Role className='ui tiny error message' role='alert'>
       <div className='header'>Something went wrong:</div>
-      <pre>{error?.message}</pre>
+      {showError && <pre>{error?.message}</pre>}
+      <br />
       <button
         className='ui negative button'
-        onClick={resetErrorBoundary}
+        onClick={hangleTryAgainClick}
         type='button'
       >
         Try again
       </button>
-    </div>
+      <label className='ui' htmlFor='chkAlwaysShowErrors'>
+        <Checkbox
+          checked={checked}
+          id='chkAlwaysShowErrors'
+          onChange={hangleCheckboxClick}
+        />{' '}
+        Always show errors
+      </label>
```

Comment:
@lordpixel what use case do we feel we are solving with this checkbox? We do not want end users to see this to be able to opt in to messages. That will basically make this checkbox show for everyone instead of the error message in production, which we don't want. We don't want customers opting in to these types of errors on screen. At least not by default. If we show this to everyone, I think we will just have everyone opting in to these everywhere.

I like the addition to have a localstorage value to enable if we needed to look at something in production, but having UI to do that I don't think is the goal here.

Since this checkbox was not in scope for the ticket description, I want to make sure I am not missing anything here. Maybe you have a good use case that you think we should actually have this.

Reply:
@brianespinosa The checkbox is there because that is what I understood from the task title:
`Improve default Boundaries FallbackComponent to only show the error message in the UI during development environments with support for showing them with a localstorage key`.

The approach I decided to go with is a checkbox that controls the localStorage value. I apologize if that is not what we want. Let me remove the checkbox and the localStorage key 👍